src/ol/renderer/layerrenderer.js

```diff
@@ -151,7 +151,7 @@ ol.renderer.Layer.prototype.loadImage = function(image) {
     // listener (a noop if the listener was already registered)
     goog.asserts.assert(imageState == ol.ImageState.IDLE ||
         imageState == ol.ImageState.LOADING);
-    goog.events.listenOnce(image, goog.events.EventType.CHANGE,
+    goog.events.listen(image, goog.events.EventType.CHANGE,
```

Comment:
@elemoine I've debugged in the layer renderer and after a zoom action there is only 1 change listener (in the line before the change listener from the renderer gets added), i.e. the one from the image wms source.

Is that enough confirmation that the listener gets garbage collected or were you thinking something else to verify this?